packages/bruno-requests/src/network/system-proxy/index.ts

```diff
@@ -0,0 +1,122 @@
+import { platform } from 'node:os';
+import { ProxyConfiguration, SystemProxyDetectorOptions } from './types';
+import { WindowsProxyDetector } from './utils/windows';
+import { MacOSProxyDetector } from './utils/macos';
+import { LinuxProxyDetector } from './utils/linux';
+import { normalizeNoProxy, normalizeProxyUrl } from './utils/common';
+
+export class SystemProxyDetector {
+  private platform: string;
+  private cachedResult: ProxyConfiguration | null = null;
+  private cacheTimestamp: number = 0;
+  private cacheTimeout: number;
+  private commandTimeout: number;
+  private activeDetection: Promise<ProxyConfiguration> | null = null;
+
+  constructor(options: SystemProxyDetectorOptions = {}) {
+    this.platform = platform();
+    this.cacheTimeout = options.cacheTimeout || 60000; // 60 seconds default
```

Comment:
@bijin-bruno @lohit-bruno Do we want to implement cache here ?
I'm afraid someone will update proxy settings while debugging and come nack to bruno to test it -- all within 60 seconds

If we haven't heard any performance issues as such, I don't think we should do this, unless you see some advantage here